app/models/upload.rb

```diff
@@ -494,14 +494,7 @@ def update_secure_status(source: "unknown", override: nil)
     self.update(secure_params(mark_secure, reason, source))
 
     if secure_status_did_change && SiteSetting.s3_use_acls && Discourse.store.external?
-      begin
-        Discourse.store.update_upload_access_control(self)
-      rescue Aws::S3::Errors::NotImplemented => err
-        Discourse.warn_exception(
-          err,
-          message: "The file store object storage provider does not support setting ACLs",
-        )
-      end
```

Comment:
This should not be a concern of the `Upload` class so I've moved it into `FileStore::S3Store#update_access_control `